bigtable/client/table_admin_test.cc

```diff
@@ -690,6 +691,75 @@ TEST_F(TableAdminTest, CheckConsistencyFailure) {
 #endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
 }
 
+/**
+ * @test Verify that `bigtagble::TableAdmin::CheckConsistency` works as
+ * expected, with multiple asynchronous calls.
+ */
+TEST_F(TableAdminTest, AsyncCheckConsistencySimple) {
+  using namespace ::testing;
+  using namespace bigtable::chrono_literals;
+
+  bigtable::LimitedTimeRetryPolicy retry_policy(std::chrono::milliseconds(10));
+  bigtable::ExponentialBackoffPolicy backoff_policy;
+  std::unique_ptr<bigtable::PollingPolicy> polling_policy =
+      bigtable::internal::make_unique<bigtable::GenericPollingPolicy<>>(
+          retry_policy, backoff_policy);
+
+  bigtable::TableAdmin tested(
+      client_, "the-async-instance", std::move(retry_policy),
+      std::move(backoff_policy), std::move(polling_policy));
+  std::string expected_text = R"""(
+name: 'projects/the-project/instances/the-async-instance/tables/the-async-table'
+consistency_token: 'test-async-token'
+    )""";
+  auto mock =
+      MockRpcFactory<btproto::CheckConsistencyRequest,
+                     btproto::CheckConsistencyResponse>::Create(expected_text);
```

Comment:
You are using the default mock action, which validates the request, but does not set any parameters in the response.  The response, therefore, is always "empty", and has no results.

You need to use a custom call here, you should consider writing a test where the response returns `false` for a couple of iterations and then returns `true`.

Reply:
Sorry I does not realise that MockRpcFactory is defined in the same file. So was wondering about test fail.
Will check it and do it.
Thanks